tools/cli/factory.go

```diff
@@ -95,16 +101,47 @@ func (b *clientFactory) SDKClient(c *cli.Context, namespace string) sdkclient.Cl
 	return sdkClient
 }
 
-func (b *clientFactory) createGRPCConnection(hostPort string) *grpc.ClientConn {
+func (b *clientFactory) createGRPCConnection(c *cli.Context) *grpc.ClientConn {
+	hostPort := c.GlobalString(FlagAddress)
 	if hostPort == "" {
 		hostPort = localHostPort
 	}
 
-	connection, err := rpc.Dial(hostPort, nil)
+	var tlsClientConfig *tls.Config
+
+	certPath := c.GlobalString(FlagTLSCertPath)
+	keyPath := c.GlobalString(FlagTLSKeyPath)
+	caPath := c.GlobalString(FlagTLSCaPath)
+
+	if caPath != "" {
+		tlsConfig := config.RootTLS{
+			Frontend: config.GroupTLS{
+				Server: config.ServerTLS{
+					CertFile:          certPath,
+					KeyFile:           keyPath,
+					ClientCAFiles:     []string{caPath},
+					RequireClientAuth: false,
+				},
+				Client: config.ClientTLS{
+					RootCAFiles: []string{caPath},
+					ServerName:  "localhost",
+				},
+			},
+		}
+		provider, err := encryption.NewLocalStoreTlsProvider(&tlsConfig)
+		if err != nil {
+			log.Fatalf("error initializing TLS provider: %v", err)
+		}
+		// TODO: should be GetFrontendClientConfig(), but it's currently not working
```

Comment:
maybe doesn't work because this should be using frontend not internode cert provider?
https://github.com/temporalio/temporal/blob/1bbdee566f2a5180b5860b3bfc7bd480a3b22bc1/common/rpc/encryption/localStoreTlsFactory.go#L64

Reply:
No, that's not fixing it.

Reply:
I feel we need to refactor our TLS settings for better clarity. But that can probably be done as a separate step.